Shoko.Server/Settings/ImportSettings.cs

```diff
@@ -8,6 +8,8 @@ public class ImportSettings
         
         public List<string> VideoExtensions { get; set; } = new List<string> { "MKV", "AVI", "MP4", "MOV", "OGM", "WMV", "MPG", "MPEG", "MK3D", "M4V" };
 
+        public List<string> Exclude { get; set; } = new List<string> { "$RECYCLE.BIN", ".Recycle.Bin", ".Trash-" };
```

Comment:
Only comment here is I don't think it's only ".trash"
Do we want to constrain these as `^\$RECYCLE\.BIN^` as `$RECYCLE.BIN` won't actually match, I'll give more examples

Reply:
https://regex101.com/r/6ynWJN/1 for example on the `$RECYCLE.BIN` example.
and https://regex101.com/r/nBx5Hq/1 for `.Recycle.Bin`

Reply:
@da3dsoul thoughts?

Reply:
Thanks for picking that up. I tried changing it to: 
`public List<string> Exclude { get; set; } = new List<string> { @"\$RECYCLE\.BIN", @"\.Recycle\.Bin", @"\.Trash-" };`

In the config file its getting serialized to:
`"Exclude": [
      "\\$RECYCLE\\.BIN",
      "\\.Recycle\\.Bin",
      "\\.Trash-"
    ],`

and then deserialized to:
![image](https://user-images.githubusercontent.com/25097788/92326003-7e61af80-f081-11ea-8495-63899ec2fd58.png)

I tried to see what would happen if I edit the config and remove the extra "\" and it then fails to deserialize:
`Shoko.Server.Settings.ServerSettings.Instance.**get** returned null.`

Any ideas ? I guess one way is to try and unescape the Regex string but having to use non-standard Regex in the config file so that it converts properly is not ideal from a user perspective.

Reply:
That's fine, it's a side-effect of JSON as `\` is the escape character, The editing of the config is really mostly handled in webui anyway.

The main thing you will want to do is use `^` and `$` to anchor to the regexes.
So you want to do:
```suggestion
        public List<string> Exclude { get; set; } = new List<string> { @"^\$RECYCLE\.BIN$", "^\.Recycle\.Bin$", "^\.Trash-" };
```

Reply:
In the WorkerFileEvents_DoWork() loop we are actually testing against the full path.
![image](https://user-images.githubusercontent.com/25097788/92326837-9e946d00-f087-11ea-8f3d-35bd6fcb7def.png)
So that would not result in a match. Ill need to double check the other places to ensure its consistant. I'm not sure if there is a way to get a relative paths in WorkerFileEvents_DoWork(), quickly looking I could not see the info being passed to it.

Example >> https://regex101.com/r/rvVwdB/2

The problem is that the regex pattern we need to pass into the Regex.IsMatch function is for exmaple `\$RECYCLE\.BIN` but due to the JSON serialization/deserialization its coming through as the below:
![image](https://user-images.githubusercontent.com/25097788/92326992-ea93e180-f088-11ea-96b2-9aea91d64f6c.png)
This wont match and will be read as a backslash instead of an escape for the "$" and "."

Reply:
Had a think about this and I believe a simple String.Replace should do the trick, basically replace "\\" with "\". Cant think of any edge cases at the momment where this may cause issues.

Reply:
That shouldn't be a problem, you are looking at the escaped version in the debug tools:

```
$ cat Program.cs
using System;
using Newtonsoft.Json;

namespace aaa
{
    class Program
    {
        static void Main(string[] args)
        {
            var data = JsonConvert.SerializeObject(new { a = @"^\.RECYCLE\.BIN^" });
            Console.WriteLine(data);

            dynamic data1 = JsonConvert.DeserializeObject("{\"a\":\"^\\\\.RECYCLE\\\\.BIN^\"}"); // {"a":"^\\.RECYCLE\\.BIN^"} just escaped.
            Console.WriteLine(data1.a);

            dynamic deser = JsonConvert.DeserializeObject(data);
            Console.WriteLine(deser.a);
        }
    }
}
$ dotnet run
{"a":"^\\.RECYCLE\\.BIN^"}
^\.RECYCLE\.BIN^
^\.RECYCLE\.BIN^
```

Reply:
ah! I dont even know.. swear I tested it and it wasn't matching but now its working. Thanks for that, will have to keep it in mind when using the debug tools. Its been a while since I touched visual studio.

Reply:
Hmm, I would still prefer to match by the folder explicitly, just trying to work out a good idea for that.

Reply:
What about this ? >> `(\\|\/)\$RECYCLE\.BIN(\\|\/)`
https://regex101.com/r/rvVwdB/3

This will keep it to folder only and match linux,windows,cifs/smb paths

Reply:
After a bit of thought, leaving as full path allows better extensibility, so we can leave as is.